src/psykal/psy.f90

```diff
@@ -118,34 +145,33 @@ subroutine zslpxy_update_psy(zslpx,zslpy,zwx,zwy,jpk,jpj,jpi)
     !
   end subroutine
   !
-  subroutine zwxy2_psy(zwx,zwy,zdt,pun,pvn,mydomain,zind,zslpx,zslpy,jpk,jpj,jpi)
+  subroutine zwxy2_psy(zwx,zwy,pun,pvn,mydomain,zind,zslpx,zslpy)
     !
     use zwxy2_kern_mod, only : zwxy2_kern
     !
-    real*8, intent(out) :: zwx(:,:,:), zwy(:,:,:)
-    real*8, intent(in)  :: pun(:,:,:), pvn(:,:,:), mydomain(:,:,:), zind(:,:,:), zslpx(:,:,:), zslpy(:,:,:)
-    real*8, intent(in)  :: zdt
-    integer, intent(in) :: jpk,jpj,jpi
+    real*8, intent(out) :: zwx(jpi,jpj,jpk), zwy(jpi,jpj,jpk)
+    real*8, intent(in)  :: pun(jpi,jpj,jpk), pvn(jpi,jpj,jpk), &
+                           mydomain(jpi,jpj,jpk), zind(jpi,jpj,jpk), &
+                           zslpx(jpi,jpj,jpk), zslpy(jpi,jpj,jpk)
     ! local variables
     integer :: jk,jj,ji   
     !
     DO jk = 1, jpk-1
        DO jj = 2, jpj-1
           DO ji = 2, jpi-1
-             call zwxy2_kern(zwx,zwy,zdt,pun,pvn,mydomain,zind,zslpx,zslpy,jk,jj,ji)
+             call zwxy2_kern(zwx,zwy,pun,pvn,mydomain,zind,zslpx,zslpy,jk,jj,ji)
```

Comment:
I failed to notice this when reviewing #3 but I think that the ordering of the `jk, jj, ji` kernel arguments is counter-intuitive. Please can you change this to `ji, jj, jk` for all kernels.